### PR TITLE
HPCC-19274  Add ECLArchiveViewer

### PR DESCRIPTION
**Pull request by GordonSmith**
Add ECL Archive tree view next to ECL textbox.

Clicking on a section name will display the contents on the right.

The ECL query is displayed on the right by default. If no archive contents are found for a WU the left side "DirectoryTree" is hidden.

Signed-off-by: Jaman Brundage <jbrundage372@gmail.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [x] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by GordonSmith** [comment]
@jbrundage please review.

Note: I have opened a new issue for needed "improvements":  https://track.hpccsystems.com/browse/HPCC-23738

File: esp/src/src/Utility.ts [hunk 1]
```diff
@@ -473,6 +473,9 @@ export function resolve(hpccWidget, callback) {
         case "ECLPlaygroundWidget":
             require(["hpcc/ECLPlaygroundWidget"], doLoad);
             break;
+        case "ECLArchiveWidget":
+            require(["src/ECLArchiveWidget"], doLoad);
+            break;
         case "ECLSourceWidget":
             require(["hpcc/ECLSourceWidget"], doLoad);
             break;
```
Comment (jbrundage) [inline esp/src/src/Utility.ts]: Shouldn't this case have been removed?

**Review by jbrundage — APPROVED** [review]

File: esp/src/src/Utility.ts (hunk 1 above)
Comment (GordonSmith) [reply esp/src/src/Utility.ts]: No - I ported the code into TypeScript (but left logic alone).

**Comment by GordonSmith** [comment]
@richardkchapman this can be pulled now.

**Pull request merged**